flink-state-backends/flink-statebackend-forst/src/main/java/org/apache/flink/state/forst/ForStKeyedStateBackend.java

```diff
@@ -438,6 +449,18 @@ private <N, SV> RegisteredKeyValueStateBackendMetaInfo<N, SV> updateRestoredStat
                             + ").");
         }
 
+        if (userKeySerializer != null) {
+            TypeSerializer<UK> previousUserValueStateSerializer =
+                    restoredKvStateMetaInfo.getUserKeySerializer();
+            TypeSerializerSchemaCompatibility<UK> userKeySerializerCompatibility =
+                    restoredKvStateMetaInfo.updateUserKeySerializer(userKeySerializer);
+            if (!previousUserValueStateSerializer.equals(userKeySerializer)
```

Comment:
`previousUserValueStateSerializer` could be null.

Reply:
@Zakelly Thanks for your suggestion , I‘ve updated the PR ， PTAL